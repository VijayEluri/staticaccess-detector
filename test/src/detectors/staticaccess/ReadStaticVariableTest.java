package detectors.staticaccess;

/**
 * Assigning a static variable test.
 * @author Adam Warski (adam at warski dot org)
 */
public class ReadStaticVariableTest {
	private static String x = "b";

	@StaticIndependent
	public static void m1() {
	    String a = x;
	}

	public static void m2() {
		String a = x;
	}
}