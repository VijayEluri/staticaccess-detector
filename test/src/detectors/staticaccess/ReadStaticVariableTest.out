H C SANA: Static-independent method detectors.staticaccess.ReadStaticVariableTest.m1() accesses static variable detectors.staticaccess.ReadStaticVariableTest.x  At ReadStaticVariableTest.java:[line 12]